### PR TITLE
Quality: Parameterize SDK and test-data branch for test workflow

### PR DESCRIPTION
**Pull request by typotter**
🎟️ Fixes FF-3086 towards FF-3085

👯‍♂️ **Related PRs**
- [sdk-test-data#57](https://github.com/Eppo-exp/sdk-test-data/pull/57)
- [node-server-sdk#69](https://github.com/Eppo-exp/node-server-sdk/pull/69)
- [dot-net-server-sdk#44](https://github.com/Eppo-exp/dot-net-server-sdk/pull/44)

### Motivation
Changes are often made to the [sdk-test-data](https://github.com/Eppo-exp/sdk-test-data) repository to capture new behaviours, bugs and edge cases. When these changes are pushed to `main`, the SDKs are cloned locally (locally to the github action running) and their respective tests are run. These tests are set up and run by copies of the SDK test workflows - see [sdk-test-data workflow](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows/test-sdks.yml). There are a number of limitations to this setup:

- Test steps are copied from the SDK’s respective workflows; changes to the SDK test workflows need to be replicated in sdk-test-data and this is not obvious to devs
- The SDK tests are not able to run against in-flight changes to `sdk-test-data`
- When new test-data is committed that breaks an SDK, that breakage is not surfaced in the SDK’s repository until some action triggers its testing workflow (on demand, on pull-request, etc.).

### Description of Changes
_This change_
🚀 - Each SDK's testing workflow is enhanced into a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), exposing parameters for SDK branch and the sdk-test-data branch to use in testing.
- The test workflow runs using the main branch of sdk-test-data on all main pushes and Pull Requests using the PR's branch

_External to this Change_
[sdk-test-data](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows) get two testing workflows.
1. ♻️  "Local Testing"- For all pull request changes, the "Local Testing" workflow calls the reusable SDK workflows, test results are recorded only in the sdk-test-data action. This is run using the main SDK branch and the "current" branch of workflow, i.e. the pull request branch. (SDK repo does not see/is not notified of test failures during PR lifecycle, only on push to main)
2. ♻️ 🧑‍💻 "Remote Testing" - On all pushes to the main branch of sdk-test-data, the SDK testing workflows are triggered to run within their respective repositories, alerting all subscribers of failed runs (repo is red/green).

### CODE REVIEW

**Review by leoromanovsky — APPROVED** [review]

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
+  build-and-test:
 
     runs-on: ubuntu-latest
 
     steps:
+    - name: Display Testing Details
+      run: |
+        echo "Running SDK Test using"
+        echo "Test Data: sdk-test-data@${TEST_DATA_BRANCH_NAME}"
+        echo "SDK Branch: php-sdk@${SDK_BRANCH_NAME}"
```
Comment (aarsilv) [inline .github/workflows/run-tests.yml:40]: Good thinking to do this

**Review by aarsilv — APPROVED** [review]
